// primevue/togglebutton/style/togglebuttonstyle.esm.js
import BaseStyle from 'primevue/base/style';

var css = "\n@layer primevue {\n    .p-togglebutton {\n        position: relative;\n        display: inline-flex;\n        user-select: none;\n        vertical-align: bottom;\n    }\n\n    .p-togglebutton-input {\n        cursor: pointer;\n    }\n\n    .p-togglebutton .p-button {\n        flex: 1 1 auto;\n    }\n}\n";
var classes = {
  root: function root(_ref) {
    var instance = _ref.instance,
      props = _ref.props;
    return ['p-togglebutton p-component', {
      'p-disabled': props.disabled,
      'p-highlight': instance.active
    }];
  },
  input: 'p-togglebutton-input',
  box: function box(_ref2) {
    var instance = _ref2.instance;
    return ['p-button p-component', {
      'p-button-icon-only': instance.hasIcon && !instance.hasLabel
    }];
  },
  icon: function icon(_ref3) {
    var instance = _ref3.instance,
      props = _ref3.props;
    return ['p-button-icon', {
      'p-button-icon-left': props.iconPos === 'left' && instance.label,
      'p-button-icon-right': props.iconPos === 'right' && instance.label
    }];
  },
  label: 'p-button-label'
};
var ToggleButtonStyle = BaseStyle.extend({
  name: 'togglebutton',
  css: css,
  classes: classes
});

export { ToggleButtonStyle as default };
